src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-manager-message.service.ts

```diff
@@ -6,180 +6,136 @@ import { Task } from '../models/task';
 import { ServicesModule } from './services.module';
 
 class TaskManagerMessage {
-  descr: (metadata) => string;
-  running: (metadata) => string;
-  success: (metadata) => string;
-  error: (metadata) => object;
-
-  constructor(
-    descr: (metadata) => string,
-    running: (metadata) => string,
-    success: (metadata) => string,
-    error: (metadata) => object
-  ) {
-    this.descr = descr;
-    this.running = running;
-    this.success = success;
-    this.error = error;
+  errors: (metadata) => object;
+  prefixes: string[];
+  suffix: (object) => string;
+
+  failure(metadata): string {
+    return `Failed to ${this.prefixes[0]} ${this.suffix(metadata)}`;
+  }
+
+  running(metadata): string {
+    return `${this.prefixes[1]} ${this.suffix(metadata)}`;
+  }
+
+  success(metadata): string {
+    return `${this.prefixes[2]} ${this.suffix(metadata)}`;
+  }
+
+  constructor(prefixes: string[], suffix: (metadata) => string, errors: (metadata) => object) {
```

Comment:
I think it would make more sense to replace `prefixes` with `states` and suffix with `ids`/`identification`/`unit`, since they are not always used in the beginning or end of the phrase.

This would also be reflected in the defaults variables names.